endpoint-sec/src/client.rs

```diff
@@ -103,24 +107,29 @@ impl Client {
                 let _err = catch_unwind(|| {
                     // Safety: Apple guarantees the received message is non-null and valid
                     let message = unsafe { Message::from_raw(message) };
-                    let mut client = Client::from_raw(client);
+                    let mut client = Client {
```

Comment:
Could we have a test making sure we can't smuggle the client out from the handler? I'm thinking of something like

```
Client::new(|client, ev| {
    let other_client = Client::new(|_, _| {});
    // Tear the `client` out of the `&mut` reference.
    let smuggled_client = std::mem::replace(client, other_client);
    static CLIENT_SMUGGLING: Mutex<Option<Client<'static>>> = Mutex::new(None);
    *CLIENT_SMUGGLING.lock() = Some(smuggled_client);
});
```

I'm expecting this to fail because `smuggled_client` must have a lifetime that's incompatible with 'static.

(Also, something else: We _may_ want to add a comment here pointing out the user _may_ swap out the contents of client from underneath us here).

Reply:
> I'm expecting this to fail because smuggled_client must have a lifetime that's incompatible with 'static. 

No, it's entirely possible to construct two clients with `static`, so I don't think we can do much here

Reply:
Fixed in !17